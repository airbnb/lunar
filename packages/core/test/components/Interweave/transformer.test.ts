import { shallowWithStyles } from '@airbnb/lunar-test-utils';
import transformer from '../../../src/components/Interweave/factories/transformer';

describe('transformer()', () => {
  it('replaces a tag with a link', () => {
    const a = document.createElement('a');
    a.setAttribute('href', 'http://airbnb.com');

    // @ts-ignore Need to fix types upstream
    const wrapper = shallowWithStyles(transformer(a, 'Airbnb'), true);

    expect(wrapper).toMatchSnapshot();
  });

  it('uses text content as children', () => {
    const a = document.createElement('a');
    a.setAttribute('href', 'http://airbnb.com');
    a.textContent = 'Children';

    // @ts-ignore Need to fix types upstream
    const wrapper = shallowWithStyles(transformer(a, []), true);

    expect(wrapper.prop('children')).toBe('Children');
  });

  it('does nothing for unknown element', () => {
    expect(transformer(document.createElement('div'), [])).toBeUndefined();
  });
});
