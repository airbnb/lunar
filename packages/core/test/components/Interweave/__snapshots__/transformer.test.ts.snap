// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transformer() replaces a tag with a link 1`] = `
<Link
  baseline={true}
  href="http://airbnb.com"
  openInNewWindow={true}
  withStylesClasses={
    Array [
      "link",
      "link_block",
      "link_baseline",
      "link_inverted",
      "link_muted",
      "link_disabled",
    ]
  }
>
  Airbnb
</Link>
`;
